tests/libs/estdlib/test_lists_subtraction.erl

```diff
@@ -0,0 +1,53 @@
+%
+% This file is part of AtomVM.
+%
+% Copyright 2024 Tomasz Sobkiewicz <tomasz.sobkiewicz@swmansion.com>
+%
+% Licensed under the Apache License, Version 2.0 (the "License");
+% you may not use this file except in compliance with the License.
+% You may obtain a copy of the License at
+%
+%    http://www.apache.org/licenses/LICENSE-2.0
+%
+% Unless required by applicable law or agreed to in writing, software
+% distributed under the License is distributed on an "AS IS" BASIS,
+% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+% See the License for the specific language governing permissions and
+% limitations under the License.
+%
+% SPDX-License-Identifier: Apache-2.0 OR LGPL-2.1-or-later
+%
+-module(test_lists_subtraction).
+
+-export([test/0]).
+
+test() ->
+    Tests = [
+        {[1, 2, 3], [1], [2, 3]},
+        {[1, 2, 3], [3], [1, 2]},
+        {[1, 2, 3], [5], [1, 2, 3]},
+        {[1, 2, 2, 3], [2], [1, 2, 3]},
+        {[1, 2, 3, 4], [2, 2], [1, 3, 4]},
+        {["a", 1, "b", 2], ["a", 2], [1, "b"]},
+        {[1, 2, 3], [1, 2, 3, 4, 5], []},
+        {[1, 2, 3], [], [1, 2, 3]},
+        {[], [1, 2, 3], []},
+        {[a, b, b, c], [c, b, a], [b]}
+    ],
+    ok = expect_failure(fun() -> [a, b | c] -- [a] end, badarg),
```

Comment:
I suggest avoiding direct calls to the functions we are testing using constants, since the compiler knows it and might raise a warning or fail in the future.
I suggest where possible using an external call, such as `?MODULE:sub([a, b | c], [a])` or anything like that, so the compiler will be more cautious about any constant folding or compile time warning.